src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax;
+
+public abstract class Figure implements Drawable, AreaCalculator {
+    private String name;
+    private String color;
+
+    public Figure(String name, String color) {
+        this.name = name;
+        this.color = color;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public abstract double calculateArea();
+
+    public void draw() {
+        System.out.print("Figure: " + toCamelCase(getName())
+                + ", area: " + calculateArea() + " sq. units"
+                + ", color: " + toCamelCase(getColor()));
+    }
+
+    private String toCamelCase(String input) {
```

Comment:
Remove this method, it is not necessary to cast color to camel case, but if you want so it is possible to add a field String camelCaseColor to Color enum